hazelcast/src/test/java/com/hazelcast/client/statistics/ClientStatisticsTest.java

```diff
@@ -190,6 +194,39 @@ public void testStatisticsClusterReconnect() {
         getStats(client, clientEngine);
     }
 
+    @Test
+    public void testStatisticsSentImmediatelyOnClusterChange() {
+        HazelcastInstance hazelcastInstance = hazelcastFactory.newHazelcastInstance();
+
+        ClientConfig clientConfig = new ClientConfig()
+                // add IMap and ICache with Near Cache config
+                .addNearCacheConfig(new NearCacheConfig(MAP_NAME))
+                .addNearCacheConfig(new NearCacheConfig(CACHE_NAME));
+
+        clientConfig.getConnectionStrategyConfig().getConnectionRetryConfig().setClusterConnectTimeoutMillis(Long.MAX_VALUE);
+        // set the collection frequency to something really high to test that statistics are sent on cluster change
+        clientConfig.getMetricsConfig()
+                .setCollectionFrequencySeconds(1000);
+
+        HazelcastInstance clientInstance = hazelcastFactory.newHazelcastClient(clientConfig);
+        HazelcastClientInstanceImpl client = getHazelcastClientInstanceImpl(clientInstance);
+
+        ReconnectListener reconnectListener = new ReconnectListener();
+        client.getLifecycleService().addLifecycleListener(reconnectListener);
+
+        hazelcastInstance.getLifecycleService().terminate();
+        Node hazelcastNode = getNode(hazelcastFactory.newHazelcastInstance());
+
+        assertOpenEventually(reconnectListener.reconnectedLatch);
+
+        // Statistics should be sent in some time that is shorter than the period
+        assertTrueEventually(() -> {
+            Collection<ClientEndpoint> endpoints = hazelcastNode.getClientEngine().getEndpointManager().getEndpoints();
+            ClientEndpoint[] endpointArray = endpoints.toArray(new ClientEndpoint[0]);
+            assertNotNull("Statistics are not sent on cluster change", endpointArray[0].getClientStatistics());
+        }, 120);
```

Comment:
```suggestion
        });
```

Minor: Isn't this already the default value?